tests/test_platypus.py

```diff
@@ -80,11 +80,38 @@ def test_platypus_nsgaii_process_pool(two_reservoir_problem):
         algorithm.run(10)
 
 
-def test_pywr_random_generator(two_reservoir_problem):
-    """ Test PywrRandomGenerator inserts the current model configuration in to the population. """
-    generator = PywrRandomGenerator(wrapper=two_reservoir_problem)
-    algorithm = NSGAII(two_reservoir_problem.problem, population_size=10, generator=generator)
-    algorithm.initialize()
-    # Ensure the first solution in the population has variable values from the model
-    solution = algorithm.population[0]
-    np.testing.assert_allclose(solution.variables, np.zeros(12))
+class TestPywrRandomGenerator:
+    def test_current_model(self, two_reservoir_problem):
+        """ Test PywrRandomGenerator inserts the current model configuration in to the population. """
+        generator = PywrRandomGenerator(wrapper=two_reservoir_problem)
+        algorithm = NSGAII(two_reservoir_problem.problem, population_size=10, generator=generator)
+        algorithm.initialize()
+        # Ensure the first solution in the population has variable values from the model
+        solution = algorithm.population[0]
+        np.testing.assert_allclose(solution.variables, np.zeros(12))
+
+    @pytest.mark.parametrize('use_current', [True, False])
+    def test_other_solutions(self, two_reservoir_problem, use_current):
+        """Test PywrRandomGenerator inserts other solutions into the population. """
+
+        # Create some alternative initial solutions
+        solutions = [
+            {
+                'control_curve': {'doubles': [1] * 12}
+            },
+            {
+                'control_curve': {'doubles': [2] * 12}
```

Comment:
Might be worth extending this so it tests setting integer variables

Reply:
I just looked at doing this. We currently have no parameters with `integer_size > 0`. So we'll have to park this.